webapp/graphite/finders/remote.py

```diff
@@ -14,134 +11,42 @@
-        # other stuff that always happens.
-        thread_count = settings.POOL_WORKERS_PER_BACKEND * len(settings.CLUSTER_SERVERS) + settings.POOL_WORKERS
-        return get_pool(name="remote_finder", thread_count=thread_count)
-
-    def find_nodes(self, query):
-        start = time.time()
-        jobs = []
-        random.shuffle(self.remote_stores)
-        for store in self.remote_stores:
-            if store.available:
-                jobs.append((store.find, query))
+            finders.append(cls(host))
+        return finders
 
-        queue = pool_apply(self.worker_pool(), jobs)
-
-        timeout = settings.REMOTE_FIND_TIMEOUT
-        deadline = start + timeout
-        done = 0
-        total = len(jobs)
-
-        while done < total:
-            wait_time = deadline - time.time()
-            nodes = []
-
-            try:
-                nodes = queue.get(True, wait_time)
-
-            # ValueError could happen if due to really unlucky timing wait_time
-            # is negative.
-            except (Queue.Empty, ValueError):
-                if time.time() > deadline:
-                    log.debug("Timed out in find_nodes after %fs" % timeout)
-                    break
-                else:
-                    continue
-
-            log.debug("Got a remote find result after %fs" % (time.time() - start))
-            done += 1
-            for node in nodes or []:
-                yield node
-
-        log.debug("Got all remote find results in %fs" % (time.time() - start))
-
-    def fetch(self, nodes_or_patterns, start_time, end_time, now=None, requestContext=None):
-        # Go through all of the remote nodes, and launch a fetch for each one.
-        # Each fetch will take place in its own thread, since it's naturally
-        # parallel work.
-
-        patterns = []
-
-        for v in nodes_or_patterns:
-            if isinstance(v, basestring):
-                patterns.append(v)
-            else:
-                patterns.append(v.path)
-
-        results = []
-        for store in self.remote_stores:
-            reader = RemoteReader(
-                store, {'intervals': []}, bulk_query=patterns)
-            result = reader.fetch_list(start_time, end_time, now, requestContext)
-            results.append(result)
-
-        def _extract():
-            for result in results:
-                result = wait_for_result(result)
-                for series in result:
-                    yield {
-                        'pathExpression': series.get('pathExpression', series['name']),
-                        'name': series['name'],
-                        'time_info': (series['start'], series['end'], series['step']),
-                        'values': series['values'],
-                    }
-
-        return FetchInProgress(_extract)
-
-
-class RemoteStore(object):
-
-    def __init__(self, finder, host):
-        self.finder = finder
+    def __init__(self, host=None):
```

Comment:
is there a specific reason to allow `host` to be `None`? doesn't that make the whole class quite useless